include/splay.h

```diff
@@ -153,13 +153,13 @@ SplayNode<V>::remove(Value const dataToRemove, SPLAYCMP * compare)
     if (splayLastResult == 0) { /* found it */
         SplayNode<V> *newTop;
 
-        if (result->left == NULL) {
+        if (result->left == nullptr) {
```

Comment:
> [Francesco](https://github.com/squid-cache/squid/pull/1075#pullrequestreview-1022406459): It highlights a lot of "pointer == nullptr" or "pointer != nullptr" idioms that feel a bit weird, but done is better than perfect so I'd land it and worry about them later on

I agree. And there were dozens of such bad expressions before this PR (almost 2000 if you count their NULL versions!). We are not making things _worse_ here by introducing something that did not exist before...

Some day, we will find and use a tool that will safely eliminate these expressions, but I doubt we should wait (may take too long while removing NULLs has immediate and long-awaited benefits) or do it now with a dumb script (too dangerous?). Said that, if others are sure that we should remove most of such expressions in this PR (to reduce the number of code changes over time), then I will adjust the PR to do it (with a dumb script).

Reply:
I am happy with this to go in, as highlighted by me approving the PR